### PR TITLE
Remove legacy `analysis.interface`, rename classes to PascalCase and rename `with_*` to `from_*`

### PR DESCRIPTION
**Pull request by DanielYang59**
### Summary

- Remove legacy `analysis.interface` (deprecated 3 years ago, see [discussion](https://github.com/materialsproject/pymatgen/pull/3702#issuecomment-2016370354))
- Rename several classes to PascalCase (see [discussion](https://github.com/materialsproject/pymatgen/pull/3702#issuecomment-2016363666))
- Rename several `with_*` classmethods to `from_*` in `analysis.graphs`

### Note:

The renamed classes in `cp2k.inputs` does not have corresponding unit tests at all.

## Summary by CodeRabbit

- **Refactor**
	- Enhanced graph initialization methods across various modules for improved consistency and future maintenance.
- **Bug Fixes**
	- Corrected class name spelling in the quasiharmonic module for clarity.
- **Documentation**
	- Added deprecation warnings for renamed methods and classes to guide developers towards updated usage.
- **Style**
	- Standardized naming conventions in input and set modules for better readability and consistency.

### CODE REVIEW

File: pymatgen/analysis/quasiharmonic.py [hunk 1]
```diff
@@ -361,8 +361,7 @@ def get_summary_dict(self):
 
 @deprecated(
     replacement=QuasiHarmonicDebyeApprox,
-    message="Use PascalCased QuasiHarmonicDebyeApprox instead. "
```
Comment (DanielYang59) [inline pymatgen/analysis/quasiharmonic.py:364]: `monty.dev.deprecated` would compile the `replacement` itself:
```python
    def craft_message(
        old: Callable,
        replacement: Callable,
        message: str,
        deadline: datetime,
    ):
        msg = f"{old.__name__} is deprecated"


        if deadline is not None:
            msg += f", and will be removed on {deadline:%Y-%m-%d}\n"


        if replacement is not None:
            if isinstance(replacement, property):
                r = replacement.fget
            elif isinstance(replacement, (classmethod, staticmethod)):
                r = replacement.__func__
            else:
                r = replacement
            msg += f"; use {r.__name__} in {r.__module__} instead."


        if message:
            msg += "\n" + message
        return msg
```
So I assume there is no need to re-declare this.

**Comment by janosh** [comment]
bear in mind that we need deprecations for all these changes. might be easier to them alongside each change rather than all at once at the end to ensure we don't forget any

File: pymatgen/analysis/quasiharmonic.py (hunk 1 above)
Comment (janosh) [reply pymatgen/analysis/quasiharmonic.py:364]: sure, feel free to change

**Comment by DanielYang59** [comment]
> bear in mind that we need deprecations for all these changes. might be easier to them alongside each change rather than all at once at the end to ensure we don't forget any

Agreed :) Already moved `replacement` and `old` together in 0a2461b1a26f73d0ec3d0b367f7e0d2a27f28be9. Thanks for the head up!

File: pymatgen/analysis/graphs.py [hunk 2]
```diff
@@ -1729,6 +1770,14 @@ def with_local_env_strategy(cls, molecule, strategy) -> Self:
         mg.set_node_attributes()
         return mg
 
+    @deprecated(
+        from_local_env_strategy,
+        "Deprecated on 2024-03-29, to be removed on 2025-03-20.",
+    )
+    @classmethod
+    def with_local_env_strategy(cls, *args, **kwargs):
```
Comment (janosh) [inline pymatgen/analysis/graphs.py:1778]: maybe try swapping the decorator order so that `classmethod` is at the top

Comment (DanielYang59) [reply pymatgen/analysis/graphs.py:1778]: Thanks. Everything looking good now.

**Comment by DanielYang59** [comment]
Please review @janosh. Thanks!

**Review by janosh — APPROVED** [review]
thanks again @DanielYang59! 👍

**Pull request merged**